src/Symfony/Component/Console/Application.php

```diff
@@ -551,23 +551,8 @@ public function addCommand(callable|Command $command): ?Command
     {
         $this->init();
 
-        if (!$command instanceof Command) {
-            if (!\is_object($command) || $command instanceof \Closure) {
-                throw new InvalidArgumentException(\sprintf('The command must be an instance of "%s" or an invokable object.', Command::class));
-            }
-
-            /** @var AsCommand $attribute */
-            $attribute = ((new \ReflectionObject($command))->getAttributes(AsCommand::class)[0] ?? null)?->newInstance()
-                ?? throw new LogicException(\sprintf('The command must use the "%s" attribute.', AsCommand::class));
-
-            $command = (new Command($attribute->name))
-                ->setDescription($attribute->description ?? '')
-                ->setHelp($attribute->help ?? '')
-                ->setCode($command);
-
-            foreach ($attribute->usages as $usage) {
-                $command->addUsage($usage);
-            }
+        if (is_callable($command)) {
```

Comment:
The command can both extend the `Command` class and implement the `__invoke` method. You need to keep the previous condition in order to fix the broken test.

```suggestion
        if (!$command instanceof Command) {
```

Reply:
Fixed.